server/src/org/apache/cloudstack/affinity/AffinityGroupServiceImpl.java

```diff
@@ -176,25 +176,31 @@ private void verifyAccessToDomainWideProcessor(Account caller, AffinityGroupProc
         }
     }
 
-    private AffinityGroupVO createAffinityGroup(final AffinityGroupProcessor processor, final Account owner, final ACLType aclType, final String affinityGroupName, final String affinityGroupType, final String description) {
+    private AffinityGroupVO createAffinityGroup(final AffinityGroupProcessor processor, final Account owner, final ACLType aclType, final String affinityGroupName, final String affinityGroupType, final String description, boolean domainLevel, Long domainId) {
+
+        Long domainIdBasedOnDomainLevel = owner.getDomainId();
```

Comment:
@GabrielBrascher what about extracting lines 181-184 to a method and then create a test case for it? 
Also a documentation describing the method seems a good idea.